File: 1800_maximum_ascending_subarray_sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& x) {
        int f = x.size(), y = 0, t = x[0];
        for(int i = 1; i < f; i++) {
            if(x[i] > x[i-1]) t += x[i];
            else y = max(y, t), t = x[i];
        }
        return max(y, t);
    }
};

File: 1800_maximum_ascending_subarray_sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int maxSum = 0, sum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                sum += nums[i];
            } else {
                maxSum = Math.max(maxSum, sum);
                sum = nums[i];
            }
        }
        return Math.max(maxSum, sum);
    }
}
